x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -69,6 +78,67 @@ const analysisFieldsWarningMessage = {
   heading: analysisFieldsHeading,
 };
 
+function getRegressionAndClassificationMessage(
+  analysisConfig: AnalysisConfig,
+  analysisType: string,
+  totalDocs: number,
+  depVarCardinality: number | undefined
+) {
+  const messages = [];
+  if (isRegressionAnalysis(analysisConfig) || isClassificationAnalysis(analysisConfig)) {
+    const trainingPercent = analysisConfig[analysisType].training_percent;
+    const featureImportance = analysisConfig[analysisType].num_top_feature_importance_values;
+    // @ts-ignore num_top_classes doesn't exist on Regression type
+    const topClasses: number | undefined = analysisConfig[analysisType].num_top_classes;
+
+    if (trainingPercent) {
+      const trainingDocs = totalDocs * (trainingPercent / 100);
+      const trainingPercentMessage = getTrainingPercentMessage(trainingDocs);
+      if (trainingPercentMessage) {
+        messages.push(trainingPercentMessage);
+      }
+    }
+
+    if (featureImportance && featureImportance > 0) {
+      messages.push({
+        id: 'feature_importance',
+        text: i18n.translate(
+          'xpack.ml.models.dfaValidation.messages.featureImportanceWarningMessage',
+          {
+            defaultMessage:
+              'Enabling feature importance can result in long running jobs for jobs with a large number of training documents.',
```

Comment:
I think it reads better if we remove one of the "jobs" from this sentence. For example:
```suggestion
              'Enabling feature importance can result in long running jobs when there are a large number of training documents.',
```

Reply:
Thanks for taking a look! 😄  Updated in 2d7ccc146f7e1b7e86daa70627c839f7e79e637c